test/run_envoy_bazel_coverage.sh

```diff
@@ -73,6 +73,8 @@ time "${GCOVR}" --gcov-exclude="${GCOVR_EXCLUDE_REGEX}" \
 # run.
 rm "${SRCDIR}"/test/coverage/BUILD
 
+[[ -z "${ENVOY_COVERAGE_DIR}" ]] || rsync -av "${COVERAGE_DIR}"/ "${ENVOY_COVERAGE_DIR}"
```

Comment:
Why only if the coverage dir doesn't exist? I often (locally) use `bazel.coverage` to generate reports, and will run it a number of times to validate the coverage status as I make code changes. Usually it just clobbers the previous report as currently implemented. In this PR, I think it would silently fail to update.

Reply:
`-z` just check if the env `ENVOY_COVERAGE_DIR` is set, not checking if dir exists, if you run `test/run_envoy_bazel_coverage.sh` directly, this env var is not set.